pandas/core/window/indexers.py

```diff
@@ -78,30 +78,16 @@ def get_window_bounds(
         closed: Optional[str] = None,
     ) -> Tuple[np.ndarray, np.ndarray]:
 
-        start_s = np.zeros(self.window_size, dtype="int64")
-        start_e = (
-            np.arange(self.window_size, num_values, dtype="int64")
-            - self.window_size
-            + 1
-        )
-        start = np.concatenate([start_s, start_e])[:num_values]
-
-        end_s = np.arange(self.window_size, dtype="int64") + 1
-        end_e = start_e + self.window_size
-        end = np.concatenate([end_s, end_e])[:num_values]
-
-        if center and self.window_size > 2:
-            offset = min((self.window_size - 1) // 2, num_values - 1)
-            start_s_buffer = np.roll(start, -offset)[: num_values - offset]
-            end_s_buffer = np.roll(end, -offset)[: num_values - offset]
+        if center:
+            offset = (self.window_size - 1) // 2
+        else:
+            offset = 0
 
-            start_e_buffer = np.arange(
-                start[-1] + 1, start[-1] + 1 + offset, dtype="int64"
-            )
-            end_e_buffer = np.array([end[-1]] * offset, dtype="int64")
+        end = np.arange(1 + offset, num_values + 1 + offset).astype("int64")
```

Comment:
you can add the dtype to the np.arange constructor

Reply:
Good call! Added